src/main/java/seedu/address/model/task/DateTime.java

```diff
@@ -25,12 +25,26 @@
      *
      * @param dateTime A valid date time.
      */
-    public DateTime(String dateTime) {
+    private DateTime(String dateTime) {
         requireNonNull(dateTime);
         checkArgument(isValidDateTime(dateTime), MESSAGE_CONSTRAINTS);
         this.dateTime = TimeFormatChecker.mapToLocalDateTime(dateTime);
     }
 
+    /**
+     * Constructs a {@code DateTime} by checking the input dateTime.
+     *
+     * @param dateTime A date time that can be invalid or valid.
+     * @return null if the date time if invalid, else return the {@code DateTime} object.
+     */
+    public static DateTime createDateTime(String dateTime) {
+        if (dateTime.equals("")) {
+            return null;
+        } else {
+            return new DateTime(dateTime);
+        }
+    }
```

Comment:
Want to check which test requires the usage of this. See if there is better solution.